### PR TITLE
Use errors instead of optionals for some functions

### PR DESCRIPTION
**Pull request by dominikh**

### CODE REVIEW

File: src/render/renderer.zig [hunk 1]
```diff
@@ -31,13 +31,21 @@ pub const Renderer = extern struct {
     pub const clear = wlr_renderer_clear;
 
     extern fn wlr_renderer_init_wl_display(r: *Renderer, server: *wl.Server) bool;
-    pub const initServer = wlr_renderer_init_wl_display;
+    pub fn initServer(r: *Renderer, server: *wl.Server) !void {
+        if (!wlr_renderer_init_wl_display(r, server)) {
+            return error.Failure;
```
Comment (ifreund) [inline src/render/renderer.zig:36]: I think we should be a little less generic with the error messages.
```suggestion
            return error.RendererInitFailure;
```

File: src/types/xcursor_manager.zig [hunk 2]
```diff
@@ -15,13 +15,19 @@ pub const XcursorManager = extern struct {
...
+    }
 
     extern fn wlr_xcursor_manager_destroy(manager: *XcursorManager) void;
     pub const destroy = wlr_xcursor_manager_destroy;
 
     extern fn wlr_xcursor_manager_load(manager: *XcursorManager, scale: f32) bool;
-    pub const load = wlr_xcursor_manager_load;
+    pub fn load(manager: *XcursorManager, scale: f32) !void {
+        if (!wlr_xcursor_manager_load(manager, scale)) {
+            return error.Failure;
```
Comment (ifreund) [inline src/types/xcursor_manager.zig:28]:
```suggestion
            return error.XcursorLoadFailure;
```

**Review by ifreund — COMMENTED** [review]
The tinywl code looks a *lot* cleaner with these changes :)

**Comment by dominikh** [comment]
PTAL. I force-pushed with your suggestions applied.

**Review by ifreund — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**